src/main/java/com/healthapp/project/Service/MedicationService.java

```diff
@@ -0,0 +1,125 @@
+                if (maybeProducts.isEmpty()) {
+                    continue;
+                }
+                // if we're here that means products are good to go
+                JsonArray products = maybeProducts.get();
+
+                Optional<String> maybeBrandName =
+                        Optional.ofNullable(
+                                products
+                                        .getAsJsonArray()
+                                        .get(0)
+                                        .getAsJsonObject()
+                                        .getAsJsonPrimitive("brand_name")
+                                        .getAsString());
+
+                Optional<String> maybeStrength =
+                        Optional.ofNullable(
+                                products
+                                        .getAsJsonArray()
+                                        .get(0)
+                                        .getAsJsonObject()
+                                        .getAsJsonArray("active_ingredients")
+                                        .getAsJsonArray()
+                                        .get(0)
+                                        .getAsJsonObject()
+                                        .getAsJsonPrimitive("strength")
+                                        .getAsString());
+
+                Optional<String> maybe_dosage_form =
+                        Optional.ofNullable(
+                                products
+                                        .getAsJsonArray()
+                                        .get(0)
+                                        .getAsJsonObject()
+                                        .getAsJsonPrimitive("dosage_form")
+                                        .getAsString());
+                if (maybeBrandName.isEmpty() || maybeStrength.isEmpty() || maybe_dosage_form.isEmpty()) {
+                    continue;
+                }
+
+                // i should have values for everything by time i'm here
+                String brandName = maybeBrandName.get();
+
+                String strength = maybeStrength.get();
+
+                String dosage_form = maybe_dosage_form.get();
+
+
+                MedicationDex medication =
+                        new MedicationDexBuilder()
+                                .brand_name(brandName)
+                                .strength(strength)
+                                .dosage_form(dosage_form)
+                                .build();
+
+                try {
+                    String statement = "INSERT INTO medications (brand_name, strength, dosage_form) VALUES (?,?,?)";
+                    jdbcTemplate.update(statement, brandName, strength, dosage_form);
+
+                } catch (Exception e) {
+                    System.out.println(e.getMessage());
+                    return false;
+                }
+
+            }
+        }
+        return true;
+    }
+}
```

Comment:
So this is pretty much the same exact code that i wrote in API caller.  I used my own medication object this time. wasn't too hard to implement since i got the logic down already.